assets/scripts/app/__tests__/NoConnectionMessage.test.js

```diff
@@ -1,44 +1,51 @@
 /* eslint-env jest */
 import React from 'react'
-import { shallow } from 'enzyme'
-import { mountWithIntl as mount } from '../../../../test/helpers/intl-enzyme-test-helper.js'
-import { NoConnectionMessage } from '../NoConnectionMessage'
+import { fireEvent } from '@testing-library/react'
+import { renderWithReduxAndIntl } from '../../../../test/helpers/render'
+import NoConnectionMessage from '../NoConnectionMessage'
 import { nonblockingAjaxTryAgain } from '../../util/fetch_nonblocking'
+import { showNoConnectionMessage } from '../../store/actions/status'
 
 jest.mock('../../util/fetch_nonblocking', () => ({
   nonblockingAjaxTryAgain: jest.fn()
 }))
 
-describe('NoConnectionMessage', () => {
-  it('renders a button', () => {
-    const wrapper = shallow(<NoConnectionMessage />)
-    expect(wrapper.find('button').length).toEqual(1)
-  })
+const initialState = {
+  status: {
+    noConnectionMessage: false
+  }
+}
 
+describe('NoConnectionMessage', () => {
   it('tries to reconnect when button is clicked', () => {
-    const wrapper = shallow(<NoConnectionMessage />)
-    wrapper.find('button').simulate('click')
+    const wrapper = renderWithReduxAndIntl(<NoConnectionMessage />, {
+      initialState
+    })
+    fireEvent.click(wrapper.getByText('Try again'))
```

Comment:
This is an example of a fairly straight-forward port from Enzyme syntax to react-testing-library syntax.

We no longer shallow-render the component. Instead, we use the `renderWithReduxAndIntl` helper function that we imported earlier. We also give it an object as its second argument that includes the `initialState` that we set up. This second argument object can be quite powerful because we can even pass in a custom Redux store, but usually, we only need to pass it the initial state.